### ISSUE TITLE
BatchNorm folding test fails for folding BN on ConvTranspose2d.

### ISSUE THREAD

**Issue by elad-c** [sony/model_optimization#1115]
### Issue Type

Bug

### Source

source

### MCT Version

nightly

### Describe the issue

```shell
Folded BN model output diverges from original model output significantly more than other BN folding tests.
```

### Expected behaviour

```shell
Same MSE between original and folded models as other layers (e.g. Conv2d).
```

### Code to reproduce the issue

```shell
replace the following lines in ()[]:

# TODO: remove this and set atol=1e-5 for all tests after fixing the issue with ConvTranspose2d.
is_convtranspose2d_in_model = nn.ConvTranspose2d in [type(module) for name, module in quant_model.named_modules()]
atol = 1e-3 if is_convtranspose2d_in_model else 1e-5

with:

atol=1e-5
```
